nixos/tests/matrix/conduwuit.nix

```diff
@@ -0,0 +1,94 @@
+import ../make-test-python.nix (
+  { pkgs, lib, ... }:
+  let
+    name = "conduwuit";
+  in
+  {
+    name = "conduwuit";
+
+    nodes = {
+      conduwuit = args: {
+        services.conduwuit = {
+          enable = true;
+          settings.global = {
+            server_name = name;
+            address = "0.0.0.0";
+            allow_registration = true;
+            yes_i_am_very_very_sure_i_want_an_open_registration_server_prone_to_abuse = true;
+          };
+          extraEnvironment.RUST_BACKTRACE = "yes";
+        };
+        networking.firewall.allowedTCPPorts = [ 6167 ];
+      };
+      client =
+        { pkgs, ... }:
+        {
+          environment.systemPackages = [
+            (pkgs.writers.writePython3Bin "do_test" { libraries = [ pkgs.python3Packages.matrix-nio ]; } ''
+              import asyncio
+
+              from nio import AsyncClient
+
+
+              async def main() -> None:
```

Comment:
Not sure how well you can debug this in case of an error

Reply:
Thanks, I'm updating the test with inspiration from other NixOS Matrix tests (mautrix-meta-postgres and appservice-irc), which log the Matrix responses and assert the respones type to not be an error.